src/status_im2/contexts/shell/events.cljs

```diff
@@ -29,59 +30,80 @@
-                                         :clock now})}
+        {:navigate-from :chats-stack
+         :card-id       id
+         :switcher-card {:type      shell.constants/private-group-chat-card
+                         :card-id   id
+                         :clock     now
+                         :screen-id id}}
 
         constants/community-chat-type
-        {:shell/navigate-from-shell-fx :communities-stack
-         :db                           (assoc-in
-                                        db
-                                        [:shell/switcher-cards (:community-id chat)]
-                                        {:type       shell.constants/community-channel-card
-                                         :id         (:community-id chat)
-                                         :clock      now
-                                         :channel-id (:chat-id chat)})}
+        {:navigate-from :communities-stack
+         :card-id       (:community-id chat)
+         :switcher-card {:type      shell.constants/community-channel-card
+                         :card-id   (:community-id chat)
+                         :clock     now
+                         :screen-id (:chat-id chat)}}
 
         nil))
 
     :community
-    {:shell/navigate-from-shell-fx :communities-stack
-     :db                           (assoc-in
-                                    db
-                                    [:shell/switcher-cards (:community-id id)]
-                                    {:type       shell.constants/community-card
-                                     :id         (:community-id id)
-                                     :clock      now
-                                     :channel-id nil})}
-
+    {:navigate-from :communities-stack
+     :card-id       (:community-id id)
+     :switcher-card {:type      shell.constants/community-card
+                     :card-id   (:community-id id)
+                     :clock     now
+                     :screen-id (:community-id id)}}
     nil))
 
+(rf/defn add-switcher-card
+  {:events [:shell/add-switcher-card]}
+  [{:keys [db now] :as cofx} view-id id]
+  (let [card-data     (calculate-card-data db now view-id id)
+        switcher-card (:switcher-card card-data)]
+    (when card-data
+      (rf/merge
+       cofx
+       {:db                           (assoc-in
+                                       db
+                                       [:shell/switcher-cards (:card-id card-data)]
+                                       switcher-card)
+        :shell/navigate-from-shell-fx (:navigate-from card-data)}
+       (switcher-cards-store/upsert-switcher-card-rpc switcher-card)))))
+
 (rf/defn close-switcher-card
   {:events [:shell/close-switcher-card]}
-  [{:keys [db]} id]
-  {:db (update-in db [:shell/switcher-cards] dissoc id)})
+  [{:keys [db] :as cofx} card-id]
+  (rf/merge
+   cofx
+   {:db (update-in db [:shell/switcher-cards] dissoc card-id)}
```

Comment:
Just `update` is sufficient instead of `update-in`.

Reply:
`update` without the square brackets around the key should work.

`(update db :shell/switcher-cards dissoc card-id)`

Reply:
oh, missed that, working now. Thank you